components/dashboard/src/utils.ts

```diff
@@ -149,3 +151,54 @@ export function storageAvailable(type: "localStorage" | "sessionStorage"): boole
         );
     }
 }
+
+type EventMap = Record<string, any[]>;
+export class ReplayableEventEmitter<EventTypes extends EventMap> extends EventEmitter {
```

Comment:
This is very useful for us here, because:
- Log emitters do not replay by default
- Xterm needs to be removed from the DOM and again re-inserted ad hoc, so its event listener should always include historical data.